BHPC is failing in fused jobs

https://github.com/tenstorrent/tt-metal/actions/runs/25957368504/job/76306743342
```
FAILED tests/ttnn/unit_tests/operations/fused/test_batch_norm.py::test_batch_norm_tests[testing_dtype=bfloat16-momentum=0.0-eps=1.0-bias=True-training=True-weight=True-check_mean=True-check_var=False-input_shapes=torch.Size([5, 8, 32, 32])] - RuntimeError: expected scalar type BFloat16 but found Float
```

https://github.com/tenstorrent/tt-metal/pull/44541 tried to fix but failed

--- a/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
+++ b/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
@@ -244,6 +244,8 @@ def test_batch_norm_fp32(
         bias=bias_tensor,
     )
     tt_output = ttnn.to_torch(tt_output_tensor_on_device)
+    # PyTorch 2.11+ requires eps > 0; substitute a negligible positive value for the reference call.
+    ref_eps = eps if eps > 0 else 1e-12
     torch_result = torch.nn.functional.batch_norm(
         input=in_data,
         running_mean=mean_data,
@@ -251,7 +253,7 @@ def test_batch_norm_fp32(
         weight=weight_data,
         bias=bias_data,
         training=training,
-        eps=eps,
+        eps=ref_eps,
     )
     assert_numeric_metrics(torch_result, tt_output, pcc_threshold=0.99, rtol=1e-3, atol=1e-6, frobenius_threshold=0.05)
 

--- a/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
+++ b/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
@@ -98,8 +98,15 @@ def test_batch_norm_tests(
     # dummy for the missing stat in the reference call. Assertions still only cover the stats
     # that were actually generated.
     channels = input_shapes[1]
-    ref_mean_data = mean_data if mean_data is not None else (torch.zeros(channels) if var_data is not None else None)
-    ref_var_data = var_data if var_data is not None else (torch.ones(channels) if mean_data is not None else None)
+    ref_dtype = in_data.dtype
+    ref_mean_data = (
+        mean_data
+        if mean_data is not None
+        else (torch.zeros(channels, dtype=ref_dtype) if var_data is not None else None)
+    )
+    ref_var_data = (
+        var_data if var_data is not None else (torch.ones(channels, dtype=ref_dtype) if mean_data is not None else None)
+    )
 
     torch_result = torch.nn.functional.batch_norm(
         input=in_data,
@@ -312,8 +319,15 @@ def test_batch_norm(input_shapes, training, check_mean, check_var, weight, bias,
     # dummy for the missing stat in the reference call. Assertions still only cover the stats
     # that were actually generated.
     channels = input_shapes[1]
-    ref_mean_data = mean_data if mean_data is not None else (torch.zeros(channels) if var_data is not None else None)
-    ref_var_data = var_data if var_data is not None else (torch.ones(channels) if mean_data is not None else None)
+    ref_dtype = in_data.dtype
+    ref_mean_data = (
+        mean_data
+        if mean_data is not None
+        else (torch.zeros(channels, dtype=ref_dtype) if var_data is not None else None)
+    )
+    ref_var_data = (
+        var_data if var_data is not None else (torch.ones(channels, dtype=ref_dtype) if mean_data is not None else None)
+    )
 
     torch_result = torch.nn.functional.batch_norm(
         input=in_data,

--- a/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
+++ b/tests/ttnn/unit_tests/operations/fused/test_batch_norm.py
@@ -709,10 +709,22 @@ def test_batch_norm_mixed_precision(
     var_ref = var_data.float() if var_data is not None else None
     weight_ref = weight_data.float() if weight_data is not None else None
     bias_ref = bias_data.float() if bias_data is not None else None
+    # PyTorch 2.11+ requires running_mean and running_var to both be None or both not-None.
+    # Supply a neutral dummy for the missing stat; assertions still only check the stats actually
+    # generated.
+    channels_ref = input_shapes[1]
+    if mean_ref is None and var_ref is not None:
+        mean_ref_call = torch.zeros(channels_ref, dtype=torch.float32)
+    else:
+        mean_ref_call = mean_ref
+    if var_ref is None and mean_ref is not None:
+        var_ref_call = torch.ones(channels_ref, dtype=torch.float32)
+    else:
+        var_ref_call = var_ref
     torch_result = torch.nn.functional.batch_norm(
         input=in_ref,
-        running_mean=mean_ref,
-        running_var=var_ref,
+        running_mean=mean_ref_call,
+        running_var=var_ref_call,
         weight=weight_ref,
         bias=bias_ref,
         training=training,

